modules/netweaver/main.tf

```diff
@@ -0,0 +1,134 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+terraform {
+  required_version = "~> 0.11.0"
+}
+
+locals {
+  device_name_1 = "${var.instance_name}-${var.device_1}"
+  device_name_2 = "${var.instance_name}-${var.device_2}"
+  device_name_3 = "${var.instance_name}-${var.device_3}"
+
+  access_config = {
+    "false" = []
+    "true"  = [{}]
+  }
+}
+
+resource "google_compute_disk" "gcp_nw_pd_0" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-0"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.usr_sap_size}"
+}
+
+resource "google_compute_disk" "gcp_nw_pd_1" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-1"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.sap_mnt_size}"
+}
+
+resource "google_compute_disk" "gcp_nw_pd_2" {
+  project = "${var.project_id}"
+  name    = "${var.instance_name}-nw-2"
+  type    = "${var.disk_type}"
+  zone    = "${var.zone}"
+  size    = "${var.swap_size}"
+}
+
+resource "google_compute_attached_disk" "gcp_nw_attached_pd_0" {
+  project     = "${var.project_id}"
+  device_name = "${var.usr_sap_size > 0 ? local.device_name_1 : ""}"
```

Comment:
What is the intent of this logic? It seems like it's meant to ignore the disks if the associated size is 0, but disks can not be created with a size of 0 so that case will not occur. If this is the intent, this logic should be replaced with logic like `count = "${var.usr_sap_size > 0 ? 1 : 0}"` added to the `google_compute_disk` resources and the `google_compute_attached_disk` resources.

Reply:
Hi Aaron, This logic was provided to us by Bohden since it was the python scripts which dynamically sets the size of the disk as per the machine type

Reply:
its a conditional expression and not a loop to introduce count factor.

Reply:
I just tested this. `google_compute_disk` does in fact accept a size of `0`, but when it is created, the size is set to 500 GB, which means that specifying a size of 0 is not an effective method for disabling the extra disks. Please adjust the logic as I suggested. Attempting to directly copy logic from the Python script to the Terraform configuration is a dubious strategy.

Reply:
Hi Aaron,
The device name is needed for the Netweaver logic to work , if I use the count instead of device name how am i supposed to get it back,

Reply:
Also I removed the pd-standard disk size variable and currently allocated its size as per usr_sap_size,sap_mnt_size as wel as swap_size so you won't be seeing that 500GB size.

Reply:
Thanks Aaron for the wonderful solution . Implemented it.